src/controllers/posts.ts

```diff
@@ -0,0 +1,54 @@
+import { RequestHandler } from 'express';
+import { Post } from "../models/posts";
+// const users = require('../data/users');
+const posts = require('../../data/posts');
+
+const POSTS: Post[] = posts;
+
+export const getPosts: RequestHandler = (req, res, next) => {
+    res.json({ posts: POSTS })
+}
+
+export const getPostById: RequestHandler<{pid: number}> = (req, res, next) => {
+    const postId = req.params.pid
+    const foundPost = POSTS.find(post => post.pid === Number(postId))
+
+    res.status(201).json({message: 'Success!', post: foundPost})
+}
+
+export const updatePost: RequestHandler<{pid: number}> = (req, res, next) => {
+    const postId = req.params.pid
+    const updatedPost = (req.body as {uid: number, title: string, content: string })
+    const postIndex = POSTS.findIndex(post => post.pid === Number(postId))
+
+    if (postIndex < 0) {
+        throw new Error('Could not find post.')
+    }
+
+    // todo ==> is there a way to pass these four params in as an object?
+    POSTS[postIndex] = new Post(POSTS[postIndex].pid, updatedPost.uid, updatedPost.title, updatedPost.content)
```

Comment:
Short answer, yes!

Your post constructor can look like
```ts
class Post {
  constructor(opts: PostArgs) {
    this.pid = opts.pid;
    this.uid = opts.uid;
    ...
  }
}
```

And you should be able to do
```ts
new Post({
  pid: 123,
  uid: 321,
  ...
});
```